javascript/index.js

```diff
@@ -87,23 +91,51 @@ function signature_valid(tlv){
     return merkle_nodes[0]
 }
 // console.log(tlv_offer[8][1](['1000']))
-function decode(paymentRequest){
-    if (typeof paymentRequest !== 'string') throw new Error('Lightning Payment Request must be string')
-    // FIXME: Test vectors
-    paymentRequest=paymentRequest.replace('+','')
-    paymentRequest=paymentRequest.replace(' ','')
+console.log(decode('lno1qcp4256ypqpq86q2pucnq42ngssx2an9wfujqerp0y2pqun4wd68jtn00fkxzcnn9ehhyec6qgqsz83qfwdpl28qqmc78ymlvhmxcsywdk5wrjnj36jryg488qwlrnzyjczlqsp9nyu4phcg6dqhlhzgxagfu7zh3d9re0sqp9ts2yfugvnnm9gxkcnnnkdpa084a6t520h5zhkxsdnghvpukvd43lastpwuh73k29qsy'))
+function decode(paymentReq){
+    if (typeof paymentReq !== 'string') 
+        throw new Error('Lightning Payment Request must be string');
+    let paymentRequest='';
+    for(let i=0;i<paymentReq.length;i++){
+        if(paymentReq[i]=='\n'||paymentReq[i]=='\r'){
+            paymentRequest+=' ';
+            continue;
+        }
+        else paymentRequest+=paymentReq[i];
+    }
+    for(let i=0;i<paymentRequest.length;i++){
+        if(paymentRequest[i]=='+'){
+            let s=i,e;
+            i++;
+            while(i<paymentRequest.length && paymentRequest[i]==' '){
+                i++;
+            }
+            e=i;
+            if(e==paymentRequest.length || s==0 || paymentRequest.charAt(s-1) in isBech32==false || paymentRequest.charAt(e) in isBech32==false)
+                throw new Error('Lightning Payment Request must be string')
+            paymentRequest=paymentRequest.slice(0,s)+paymentRequest.slice(e);
+        }
+    }
+    if(paymentRequest.indexOf(' ')!=-1||paymentRequest.indexOf('+')!=-1)
+        throw new Error('Lightning Payment Request must be string');
+    
+    if(paymentRequest!=paymentRequest.toLowerCase()&&paymentRequest!=paymentRequest.toUpperCase())
+        throw new Error('Lightning Payment Request must be string');
+    paymentRequest=paymentRequest.toLowerCase();
+    
+    if (paymentRequest.slice(0, 2) != 'ln') 
+        throw new Error('Not a proper lightning payment request');
 
-    // FIXME: MUST all be UPPER or all LOWER
-    for(let i=0; i<paymentRequest.length; i++){
-        if(!paymentRequest.charAt(i) in isBech32)
-            throw new Error('Not a proper lightning payment request')
+    if (paymentRequest.indexOf('1')==-1)
+        throw new Error('Separator not present');
```

Comment:
Just get the entire human readable prefix, like so:

```
hrplen = payRequest.lastIndexOf('1')
if (hrplen == -1)
     throw new Error('Separator not present')
prefix = payRequest.slice(0,hrplen)
encodeddata = payRequest.slice(hrplen+1)
```

Reply:
Yes, I did that, this is an older commit, it seems..